vms/evm/metrics/prometheus/prometheus.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package prometheus
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/ava-labs/libevm/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+var (
+	_ prometheus.Gatherer = (*Gatherer)(nil)
+
+	errMetricSkip             = errors.New("metric skipped")
+	errMetricTypeNotSupported = errors.New("metric type is not supported")
+	quantiles                 = []float64{.5, .75, .95, .99, .999, .9999}
+	pvShortPercent            = []float64{50, 95, 99}
+)
+
+// Gatherer implements the [prometheus.Gatherer] interface by gathering all
+// metrics from a [Registry].
+type Gatherer struct {
+	registry Registry
+}
+
+// Gather gathers metrics from the registry and converts them to
+// a slice of metric families.
+func (g *Gatherer) Gather() (mfs []*dto.MetricFamily, err error) {
+	// Gather and pre-sort the metrics to avoid random listings
+	var names []string
+	var errs []error
+	g.registry.Each(func(name string, _ any) {
+		names = append(names, name)
+	})
+	slices.Sort(names)
+
+	mfs = make([]*dto.MetricFamily, 0, len(names))
```

Comment:
nit: can we define `errs` closer to where it is used?
```suggestion
	var names []string
	g.registry.Each(func(name string, _ any) {
		names = append(names, name)
	})
	slices.Sort(names)

	var (
		mfs = make([]*dto.MetricFamily, 0, len(names))
		errs []error
	)
```